extensions/blocks/podcast-player/podcast-player.php

```diff
@@ -0,0 +1,192 @@
+const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
+const BLOG_SLUG    = 'jetpack-' . FEATURE_NAME;
+/**
+ * Registers the block for use in Gutenberg
+ * This is done via an action so that we can disable
+ * registration if we need to.
+ */
+function register_block() {
+	jetpack_register_block(
+		BLOCK_NAME,
+		array(
+			'attributes'      => array(
+				'url'         => array(
+					'type' => 'url',
+				),
+				'itemsToShow' => array(
+					'type'    => 'integer',
+					'default' => 5,
+				),
+			),
+			'render_callback' => __NAMESPACE__ . '\render_block',
+		)
+	);
+}
+add_action( 'init', __NAMESPACE__ . '\register_block' );
+
+/**
+ * Podcast Player block registration/dependency declaration.
+ *
+ * @param array  $attributes Array containing the Podcast Player block attributes.
+ * @param string $content    String containing the Podcast Player block content.
+ *
+ * @return string
+ */
+function render_block( $attributes, $content ) {
+
+	if ( empty( $attributes['url'] ) ) {
+		return;
+	}
+
+	$track_list = get_track_list( $attributes['url'], $attributes['itemsToShow'] );
+
+	if ( is_wp_error( $track_list ) ) {
+		return '<p>' . __( 'Unable to retrieve track list. Please check your Podcast feed URL.', 'jetpack' ) . '</p>';
+	}
+
+	return render_player( $track_list, $attributes );
+}
+
+/**
+ * Renders the HTML for the Podcast player and tracklist.
+ *
+ * @param array $track_list the list of podcast tracks.
+ * @param array $attributes Array containing the Podcast Player block attributes.
+ * @return string the HTML for the podcast player.
+ */
+function render_player( $track_list, $attributes ) {
+	global $content_width;
+
+	$player_data = array(
+		'type'         => 'audio',
+		// Don't pass strings to JSON, will be truthy in JS.
+		'tracklist'    => true,
+		'tracknumbers' => true,
+		'images'       => true,
+		'artists'      => true,
+		'tracks'       => $track_list,
+	);
+
+	$outer         = 22; // Default padding and border of wrapper.
+	$default_width = 640;
+	$theme_width   = empty( $content_width ) ? $default_width : ( $content_width - $outer );
+
+	// If there are no tracks (it is possible) then display appropriate user facing error message.
+	if ( empty( $track_list ) ) {
+		return '<p>' . __( 'No tracks available to play.', 'jetpack' ) . '</p>';
+	}
+
+	ob_start();
+	wp_playlist_scripts( 'audio' );
+	/**
+	 * Prints and enqueues playlist scripts, styles, and JavaScript templates.
+	 *
+	 * @since 3.9.0
+	 *
+	 * @param string $type  Type of playlist. Possible values are 'audio' or 'video'.
+	 * @param string $style The 'theme' for the playlist. Core provides 'light' and 'dark'.
+	 */
+	do_action( 'wp_playlist_scripts', 'audio', 'light' );
+
+	$initial_track_src = ! empty( $track_list[0]['src'] ) ? $track_list[0]['src'] : '';
+	$css_class         = Jetpack_Gutenberg::block_classes( BLOG_SLUG, $attributes );
+
+	?>
+	<div class="wp-block-<?php echo esc_attr( $css_class ); ?> wp-playlist wp-audio-playlist wp-playlist-light">
```

Comment:
I think this could be simplified quite a bit by relying on the `block_classes` method, a bit like this:

```diff
diff --git a/extensions/blocks/podcast-player/podcast-player.php b/extensions/blocks/podcast-player/podcast-player.php
index 06c1e1159..71d594ed9 100644
--- a/extensions/blocks/podcast-player/podcast-player.php
+++ b/extensions/blocks/podcast-player/podcast-player.php
@@ -14,7 +14,7 @@ use Jetpack_Gutenberg;
 
 const FEATURE_NAME = 'podcast-player';
 const BLOCK_NAME   = 'jetpack/' . FEATURE_NAME;
-const BLOG_SLUG    = 'jetpack-' . FEATURE_NAME;
+
 /**
  * Registers the block for use in Gutenberg
  * This is done via an action so that we can disable
@@ -104,10 +104,18 @@ function render_player( $track_list, $attributes ) {
 	do_action( 'wp_playlist_scripts', 'audio', 'light' );
 
 	$initial_track_src = ! empty( $track_list[0]['src'] ) ? $track_list[0]['src'] : '';
-	$css_class         = Jetpack_Gutenberg::block_classes( BLOG_SLUG, $attributes );
+	$css_class         = Jetpack_Gutenberg::block_classes(
+		FEATURE_NAME,
+		$attributes,
+		array(
+			'wp-playlist',
+			'wp-audio-playlist',
+			'wp-playlist-light',
+		)
+	);
 
 	?>
-	<div class="wp-block-<?php echo esc_attr( $css_class ); ?> wp-playlist wp-audio-playlist wp-playlist-light">
+	<div class="<?php echo esc_attr( $css_class ); ?> ">
 		<div class="wp-playlist-current-item"></div>
 		<audio src="<?php echo esc_attr( $initial_track_src ); ?>" controls="controls" preload="none" width="<?php echo esc_attr( (int) $theme_width ); ?>"></audio>
 		<div class="wp-playlist-next"></div>

```

Reply:
those playlist classes will be removed in an upcoming PR. let's do it as simply as possible in this PR

Reply:
imo the `block_classes` is the simplest way to go, and it allows us to support folks who may enter custom CSS classes in the block's sidebar.

Reply:
do we know what CSS classes are we going to use here? I can update the code using the `block_classes` helper method.

Reply:
let's just use basic `block_classes` without anything custom added

Reply:
custom ones are already added there.

```php
<div class="wp-block-<?php echo esc_attr( $css_class ); ?> wp-playlist wp-audio-playlist wp-playlist-light">
```

Reply:
The current implementation will not output the right class, though:

![image](https://user-images.githubusercontent.com/426388/76545423-0af03400-648a-11ea-9698-b9b563a16419.png)

Reply:
removed additional classes [here](https://github.com/Automattic/jetpack/pull/14948/commits/7d3183c7784802e2a7612c463b920275092f8f6e)